janus_server/src/datastore.rs

```diff
@@ -208,25 +254,35 @@ impl Error {
     }
 }
 
-#[cfg(test)]
-pub(crate) mod test_util {
+// debug_assertions allows this module to be used in integration tests.
+// XXX: if the debug_assertions strategy doesn't make it through code review, move things back to dev-dependencies
+#[cfg(any(test, debug_assertions))]
+pub mod test_util {
```

Comment:
I'd prefer we avoid conditional compilation on `--cfg debug_assertions`, note that it breaks `cargo test --release`. Two alternative solutions:

1. Make the module `#[doc(hidden)]` and accept taking testcontainers and lazy_static as runtime dependencies.
2. Move the contents of this module to a separate file, and include it in the module tree of both the library crate (behind `#[cfg(test)]` and integration crate. This will require using `#[path="..."]` on one of the two module declarations. (note that the contents of `test_util` only depend on the public API of the crate, particularly, `Datastore::new()`, so it can be made to work in both contexts)

Reply:
The other difficulty is pulling in certain dependencies only in the integration test target. In `libprio-rs`, I made a workspace and put a distinct `Cargo.toml` for the binaries so that I could enable features in crate `prio` only when building binary `generate-test-vector`:

https://github.com/divviup/libprio-rs/blob/main/binaries/Cargo.toml
```
prio = { path = "..", features = ["test-vector"] }
```

Then `test-vector` conditionally enables dependencies: https://github.com/divviup/libprio-rs/blob/main/Cargo.toml

However if we want to do that, we could wait until we adopt @divergentdave's proposed crate layout in #18.

Reply:
I went with `#[doc(hidden)]` for now. I'm not happy with it, but it's better than the other solutions IMO.

TBH, I think this is a wart on Rust. I understand the motivation of wanting integration tests to use the public API; but this feels like the wrong lever, as I'm only a moderately-experienced Rust developer and this is already the first time I've been caught off-guard by integration tests not being compiled under `#[cfg(test)]`.